service/interpreter/workflowImpl.go

```diff
@@ -784,6 +797,23 @@ func invokeStateExecute(
 			// panic will let the workflow task will retry until the signal is sent
 			panic(fmt.Errorf("failed to signal on completion %w", err))
 		}
+
+		if response.FirstRunId != "" { // Temporal
+			// Start WaitForStateCompletionWorkflow with a new name to ensure smooth transition
+			err = unifiedClient.SignalWithStartWaitForStateCompletionWorkflow(
```

Comment:
Nit: we can refactor the code to a common method

Reply:
Done